packages/react-dataparcels/src/useParcelSideEffect.js

```diff
@@ -1,43 +1,136 @@
 // @flow
 
 import type ChangeRequest from 'dataparcels/ChangeRequest';
-import type {ParcelValueUpdater} from 'dataparcels';
 
+// $FlowFixMe - useState is a named export of react
+import {useRef} from 'react';
 // $FlowFixMe - useState is a named export of react
 import {useState} from 'react';
 
+import isPromise from 'is-promise';
 import Parcel from 'dataparcels';
-import ApplyBeforeChange from './util/ApplyBeforeChange';
+
+type QueueItem = [Parcel, ChangeRequest];
+
+const mergeQueue = (parcel: Parcel, queue: QueueItem[]): QueueItem[] => {
```

Comment:
A better return type for this would be [QueueItem] seeing as it always returns a single item

Reply:
oh yeah, good point, happy with it as is.